tools/start_server.py

```diff
@@ -39,6 +39,72 @@
 import uuid
 import time
 import sys
+import socketserver
+from http.server import SimpleHTTPRequestHandler
+import threading
+
+class WebengineFileServer():
+  def __init__(self, _host, _port):
+    self.HOST = _host
+    self.PORT = _port
+    self.tcp_server = None
+    self.app_dir_mapping = {}
+
+  def start(self):
+    socketserver.TCPServer.allow_reuse_address = True
+    try:
+      self.tcp_server = socketserver.TCPServer((self.HOST, self.PORT), self.getRequestHandler(self.app_dir_mapping))
+      self.tcp_server.serve_forever()
+      self.stop()
+    except KeyboardInterrupt:
+      pass
+    except Exception as e:
+      print('Failed to open file server at port %s' % self.PORT)
+      print(e)
+      self.tcp_server = None
+
+  def stop(self):
+    if self.tcp_server is not None:
+      self.tcp_server.shutdown()
+      self.tcp_server.server_close()
+
+  def add_app_mapping(self, _secret_key, _app_dir):
+      self.app_dir_mapping[_secret_key] = _app_dir
+
+  def remove_app_mapping(self, _secret_key):
+      self.app_dir_mapping.pop(_secret_key)
+
+  @staticmethod
+  def getRequestHandler(app_dir_mapping):
+    class Handler(SimpleHTTPRequestHandler):
+      def end_headers(self):
+        self.send_header('Access-Control-Allow-Origin', '*')
+        super().end_headers()
+        
+      def do_GET(self):
+        path_parts = self.path.split('/')
+        key = path_parts[1]
+        file_path = '/'.join(path_parts[2:])  
+        print('key is %s, path is %s' % (key, file_path))
+
+        # Check if the secret key is valid 
+        if key not in app_dir_mapping:
+          super().send_error(403, "Using invalid key %s" % key)
+          return
+
+        # Get path relative to current working directory
+        app_dir_path = app_dir_mapping[key].lstrip(os.path.commonpath([app_dir_mapping[key], os.getcwd()]))
+
+        # Check if requested path is a directory
+        if os.path.isdir(os.path.join(os.getcwd(), app_dir_path, file_path)):
+          super().send_error(403, "Cannot list directories")
+          return
+
+        self.path = '/%s/%s' % (app_dir_path, file_path)
+        print(self.translate_path(self.path))
```

Comment:
i think this print can be removed

Reply:
Fixed in https://github.com/smartdevicelink/generic_hmi/pull/273/commits/7b8af7053201d5054d86aafe6805c288ed83f666